src/panels/config/entities/ha-config-entities.ts

```diff
@@ -156,18 +167,38 @@ export class HaConfigEntities extends SubscribeMixin(LitElement) {
   private _filteredEntities = memoize(
     (
       entities: EntityRegistryEntry[],
+      states: HassEntities,
       showDisabled: boolean,
-      showUnavailable: boolean
+      showUnavailable: boolean,
+      showReadOnly: boolean
     ) => {
+      let stateEntities: EntityRegistryEntry[] = [];
+      if (showReadOnly) {
+        const regEntityIds = entities.map((entity) => entity.entity_id);
+        const stateEntityIds = Object.keys(states).filter(
+          (entityId) => !regEntityIds.includes(entityId)
+        );
+        stateEntities = stateEntityIds.map((entityId) => {
```

Comment:
I think that this would be more readable.

```ts
for (const entityId of Object.keys(states)) {
  if (regEnttiyIds.has(entityId) continue;
  stateEntities.append({…})
}
```